Add dual adrv9001 zcu102 (#1621)
analogdevicesinc/hdl: docs/projects/adrv9001/index.rst

modes:

- 2R2T - 2x Rx and 2x Tx independent control and DMAs for the two RF channels
- 1R1T - 1x Rx and 1x Tx common control and DMAs for the two RF channels also
  noted as R1_MODE

For any mode the number of RF channels(two) doesn't change only the controlling
instance and the DMAs.

The design has two receive paths and two transmit paths.
One of the receive paths (Rx12) has four channels and the other (Rx2) two
channels. These only work independently, not concomitantly.
One must chose between two active paths (2R2T), or just the Rx12 (1R1T) path,
which has four active channels, while Rx2 is disabled.
The same applies to the transmit path but in the other direction.

When only the Rx12 path is active with four channels mode, the core will take
ownership of both source synchronous interfaces. The requirement in this case
is that both interfaces run at the same rate.

Regarding the related parameters INDEPENDENT_1R1T_SUPPORT and COMMON_2R2T_SUPPORT.
The purpose of this parameters is to keep only the desired resources.
By default all modes are available.

Clock scheme
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The clocks are managed by the device and are software programmable. Please
> refer to the device data sheet for the various clocks within the device.

Reviewer: ```suggestion
The clocks are managed by the device and are software programmable. Please
refer to the device datasheet for the various clocks within the device.
```

The board provides a 38.4MHz crystal for the :adi:`ADRV9002`. It can also use
an external reference clock.

CPU/Memory interconnects addresses
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The addresses are dependent on the architecture of the FPGA, having an offset
added to the base address from HDL (see more at :ref:`architecture cpu-intercon-addr`).

> AMD

Reviewer: ```suggestion
**AMD**
```

==================== =============== ===========
Instance             Zynq/Microblaze ZynqMP
==================== =============== ===========
axi_adrv9001         0x44A00000      0x84A000000
axi_adrv9001_rx1_dma 0x44A30000      0x84A300000
axi_adrv9001_rx2_dma 0x44A40000      0x84A400000
axi_adrv9001_tx1_dma 0x44A50000      0x84A500000
axi_adrv9001_tx2_dma 0x44A60000      0x84A600000

> ==================== =============== ===========

Reviewer: Addresses should be written in this format: 0x44A0_0000. Please update the table. (also for the intel table)

> Intel

Reviewer: ```suggestion
**Intel**
```
